blockmanager.go

```diff
@@ -221,6 +238,17 @@ func (b *blockManager) resetHeaderState(newestHash *wire.ShaHash, newestHeight i
 	}
 }
 
+// RelayTxDisabled returns whether or not relaying of transactions for the given
+// peer is disabled.
+// It is safe for concurrent access.
+func (b *blockManager) RelayTxDisabled(p *peer.Peer) bool {
+	pd := b.peers[p.Id()]
```

Comment:
This should be checking if the peer entry exists and warn if it doesn't.  Currently, it will just panic and take down the entire server when given a peer that is no longer in the map.  Also, I think the map access should be protected for it to be concurrent safe as well.


For the warning part, something like:

```Go
data, ok := b.peers[p.ID()]
if !ok {
	log.Warnf(....)
}
...
```